### ISSUE TITLE
linter: eslint/sort-keys group separation breaks on \r\n

### ISSUE THREAD

**Issue by Kirens** [oxc-project/oxc#21606]
### What version of Oxlint are you using?

1.61.0

### What command did you run?

`oxlint example.js`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["oxc"],
  "rules": {
    "eslint/sort-keys": ["error", "asc", { "allowLineSeparatedGroups": true }]
  }
}
```

### What happened?

The example below is a correct example of sort-keys with `allowLineSeparatedGroups`
```js
var ok = {
  c: 1,
  d: 2,

  a: 3,
  b: 4,
};
```
However, if the code is using CRLF line breaks oxlint does not identify the two separeate groups and thus complains. This is a regression from eslint.

It seems the culprit is that groups are identified by two consecutive LF-characters:
https://github.com/oxc-project/oxc/blob/43adae9e0c06770e0488d204014ad9bfd3875fae/crates/oxc_linter/src/rules/eslint/sort_keys.rs#L187
